src/pymor/parameters/interfaces.py

```diff
@@ -37,6 +37,11 @@ def evaluate(self, mu=None):
         """Evaluate the functional for the given |Parameter| `mu`."""
         pass
 
+    @abstractmethod
+    def d_mu(self, component, index=()):
+        """returns the derivative of the functional as a new ParameterFunctional."""
```

Comment:
should be extended somewhat similar to the docstring of `OperatorInterface.d_mu`